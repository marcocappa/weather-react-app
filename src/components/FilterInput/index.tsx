import React from 'react';

const FilterInput = ({
    label,
    onChange,
}: {
    label: string;
    onChange: React.ChangeEventHandler<HTMLInputElement>;
}): JSX.Element => (
    <div>
        <label>
            {label}
            <input
                type="number"
                onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
                    onChange(e)
                }
            />
        </label>
    </div>
);

export default FilterInput;
