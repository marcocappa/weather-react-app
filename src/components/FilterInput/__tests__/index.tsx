import React from 'react';
import { shallow } from 'enzyme';
import FilterInput from '../index';

describe('<FilterInput />', () => {
    const onChange = jest.fn();
    const props = {
        label: 'Test',
        onChange,
    };

    let wrapper: any;
    beforeEach(() => {
        wrapper = shallow(<FilterInput {...props} />);
    });
    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should invoke onChange', () => {
        const event = { target: { value: 10 } };
        wrapper.find('input').simulate('change', event);
        expect(onChange).toHaveBeenCalledTimes(1);
        expect(onChange).toHaveBeenCalledWith(event);
    });
});
