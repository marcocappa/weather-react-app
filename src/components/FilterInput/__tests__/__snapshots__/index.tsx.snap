// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterInput /> should match snapshot 1`] = `
<div>
  <label>
    Test
    <input
      onChange={[Function]}
      type="number"
    />
  </label>
</div>
`;
